homeassistant/components/velbus/__init__.py

```diff
@@ -85,6 +86,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: VelbusConfigEntry) -> bo
         entry.data[CONF_PORT],
         cache_dir=hass.config.path(STORAGE_DIR, f"velbuscache-{entry.entry_id}"),
     )
+
+    try:
+        await controller.connect(True)
```

Comment:
I find the re-using of `controller.connect` confusing.
Looking at the code behind connect, it feels like maybe this should be reviewed to be more explicit between connect (=NotReady), auth (=AuthFailed) and scan (in background task).

Reply:
in velbus we are have no way to know if the authentication passed or failed.
What we can do is check (after connecting) if we receive messages on the bus, but if nothing  is active this can take a really long time before we see the first message.

So not sure if we can correctly assert on a failed authentication.

I added the argument name, so its really clear that we are talking about the testing of the connection

Reply:
Does that still not mean you are connecting twice?

Reply:
yeah nice catch, i'll add a cleanup after the test

Reply:
My point is that you should be able to reuse the "prepared" connection:
connect once, then do auth+scan only in the long-running task.

Reply:
hmm, this is a change in the library, the lib is not ready for this ....

Reply:
can we dos this in a follow-up PR?
As this will also require a dependancy bump

Reply:
Since the change is only for the quality scale, I feel it is better to mark this PR as draft waiting for the dependency bump

Reply:
done